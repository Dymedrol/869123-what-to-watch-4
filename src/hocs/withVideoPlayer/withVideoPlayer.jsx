import React, {PureComponent} from "react";
import VideoPlayer from "../../components/videoPlayer/videoPlayer.jsx";

const withVideoPlayer = (Component) => {
  class WithVideoPlayer extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        isPlaying: false
      };

      this._playVideo = this._playVideo.bind(this);
      this._stopVideo = this._stopVideo.bind(this);
    }

    _playVideo() {
      this.setState({
        isPlaying: true
      });
    }

    _stopVideo() {
      this.setState({
        isPlaying: false
      });
    }

    render() {
      const {isPlaying} = this.state;
      const {isMuted} = this.props;

      return (
        <Component
          {...this.props}
          isPlaying={isPlaying}
          playVideo={() => {
            this._playVideo();
          }}
          stopVideo = {() => {
            this._stopVideo();
          }}
          isVideoPlaying={this.state.isPlaying}
          renderPlayer={(cardData) => {
            return (
              <VideoPlayer
                cardData = {cardData}
                isMuted = {isMuted}
                isPlaying={this.state.isPlaying}
              />
            );
          }}
        />
      );
    }
  }

  return WithVideoPlayer;
};

export default withVideoPlayer;
