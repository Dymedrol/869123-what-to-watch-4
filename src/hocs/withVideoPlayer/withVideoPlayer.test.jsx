import React from "react";
import renderer from "react-test-renderer";

import withVideoPlayer from "./withVideoPlayer.jsx";

describe(`Проверка снэпшота withVideoPlayer`, () => {

  const cardData = {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/a/e1/ae1ea61fe9c315fbd913c38d6ddc7c0d.480.mp4`,
  };

  const MockComponent = (props) => {
    const {
      onMouseEnter,
      onMouseLeave,
      renderPlayer,
      cardData,
    } = props;

    return (
      <article
        onMouseEnter = {onMouseEnter}
        onMouseLeave = {onMouseLeave}
        cardData = {cardData}
        isPlaying = {false}
      >
        {renderPlayer(cardData)}
      </article>
    );
  };

  // MockComponent.propTypes = {
  //   isPlaying: PropTypes.bool.isRequired,
  //   cardData: PropTypes.shape({
  //     title: PropTypes.string.isRequired,
  //     src: PropTypes.string.isRequired,
  //     preview: PropTypes.string.isRequired,
  //   }),
  // };


  const MockComponentWrapped = withVideoPlayer(MockComponent);

  it(`рендер withVideoPlayer`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        setActiveItem = {false}
        onActiveItemChange = {() => {}}
        removeActiveItem = {() => {}}
        cardData = {cardData}

      />), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});