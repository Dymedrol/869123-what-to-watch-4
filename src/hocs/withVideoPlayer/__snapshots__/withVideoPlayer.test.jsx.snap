// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка снэпшота withVideoPlayer рендер withVideoPlayer 1`] = `
<article
  cardData={
    Object {
      "preview": "https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/a/e1/ae1ea61fe9c315fbd913c38d6ddc7c0d.480.mp4",
      "src": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
      "title": "Fantastic Beasts: The Crimes of Grindelwald",
    }
  }
  isPlaying={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <video
    className="player__video"
    poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
  />
</article>
`;
